doc/conf_ref.rst

```diff
@@ -208,6 +208,23 @@ configuration file by your distribution to override the DNF defaults.
     and overwritten by \-\ :ref:`-verbose <verbose_options-label>` commandline
     option.
 
+.. _exclude_from_weak-label:
+
+``exclude_from_weak``
+    :ref:`list <list-label>`
+
+    Prevent installing packages as weak dependencies (recommends or supplements). The packages can be specified
+    by a name or a glob and separated by a comma. Defaults to ``[]``.
+
+.. _exclude_from_weak_autodetect-label:
+
+``exclude_from_weak_autodetect``
+    :ref:`boolean <boolean-label>`
+
+    If enabled, dnf will autodetect unmet weak dependencies (recommends or supplements) of installed packages and won't
+    install providers of these dependencies as weak dependencies (but they can still be installed to satisfy regular
+    dependencies). Defaults to ``true``.
```

Comment:
Does this only affect upgrade/downgrade/reinstall operations or does it also mean that, having package A installed on the system that has a weak dep W which is not installed, if I install package B which also has a weak dep W, W will not be installed?

I think this should be clarified. Otherwise the text is very precise, maybe not easy to understand at first, after the above detail is added, maybe a short "in other words" could be added too to help with understanding what it does.

Reply:
It affects all operations

Reply:
I agree we should clarify that unrelated package B also won't bring in W, since this is a serious limitation of the autodetection. So, how about this:

If enabled, dnf will autodetect unmet weak dependencies (recommends or supplements) of installed packages and won't install providers of these dependencies as weak dependencies (even if they satisfy weak dependency of another package). However, they can still be installed to satisfy regular dependencies. Defaults to ``true``.

Reply:
Fixed

Reply:
> If enabled, dnf will autodetect unmet weak dependencies (recommends or supplements) of installed packages and won't install providers of these dependencies as weak dependencies (even if they satisfy weak dependency of another package). However, they can still be installed to satisfy regular dependencies. Defaults to true.

@pkratoch while I guess it technically answers my question, it's really not obvious until you think hard about it. What about:
> If enabled, dnf will autodetect unmet weak dependencies (recommends or supplements) of packages installed on the system. Providers of these weak dependencies will not be installed by dnf as weak dependencies any more (they will still be installed if pulled in as a regular dependency).

Reply:
Thank you both for improvements

Reply:
I pushed an update